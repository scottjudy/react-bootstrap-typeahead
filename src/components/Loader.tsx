import PropTypes from 'prop-types';
import React from 'react';

const propTypes = {
  label: PropTypes.string,
};

const defaultProps = {
  label: 'Loading...',
};

interface LoaderProps {
  label?: string;
}

const Loader = ({ label }: LoaderProps) => (
  <div className="rbt-loader spinner-border spinner-border-sm" role="status">
    <span className="sr-only visually-hidden">{label}</span>
  </div>
);

Loader.propTypes = propTypes;
Loader.defaultProps = defaultProps;

export default Loader;
