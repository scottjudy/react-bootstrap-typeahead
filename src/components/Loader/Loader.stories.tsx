/* eslint-disable sort-keys,import/no-extraneous-dependencies */

import React from 'react';
import { Story, Meta } from '@storybook/react';

import Loader, { LoaderProps } from './Loader';

export default {
  title: 'Components/Loader',
  component: Loader,
} as Meta;

const Template: Story<LoaderProps> = (args) => <Loader {...args} />;

export const Default = Template.bind({});
Default.args = {};
