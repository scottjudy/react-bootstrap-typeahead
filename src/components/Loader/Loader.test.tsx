import React from 'react';

import * as stories from './Loader.stories';

import {
  composeStories,
  generateSnapshots,
  render,
  screen,
} from '../../tests/helpers';

const { Default } = composeStories(stories);

describe('<Loader>', () => {
  generateSnapshots(stories);

  it('renders a loading indicator', () => {
    render(<Default />);

    expect(screen.getByRole('status')).toHaveClass(
      'rbt-loader spinner-border spinner-border-sm'
    );
    expect(screen.getByText('Loading...')).toBeTruthy();
  });

  it('renders a custom label for accessibility', () => {
    render(<Default label="Waiting..." />);
    expect(screen.getByText('Waiting...')).toBeTruthy();
  });
});
