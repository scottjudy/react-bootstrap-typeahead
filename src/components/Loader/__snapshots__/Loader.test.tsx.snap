// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader> Default story renders snapshot 1`] = `
<div
  className="rbt-loader spinner-border spinner-border-sm"
  role="status"
>
  <span
    className="sr-only visually-hidden"
  >
    Loading...
  </span>
</div>
`;
